File: android/src/main/java/com/mycompany/capacitor/device/check/CapacitorDeviceCheckPlugin.java
package com.mycompany.capacitor.device.check;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "CapacitorDeviceCheck")
public class CapacitorDeviceCheckPlugin extends Plugin {

    private CapacitorDeviceCheck implementation = new CapacitorDeviceCheck();

    @PluginMethod
    public void generateToken(PluginCall call) {        
        JSObject ret = new JSObject();
        ret.put("token", implementation.generateToken());
        call.resolve(ret);
    }
}

File: ios/Sources/CapacitorDeviceCheckPlugin/CapacitorDeviceCheckPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorDeviceCheckPlugin)
public class CapacitorDeviceCheckPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CapacitorDeviceCheckPlugin"
    public let jsName = "CapacitorDeviceCheck"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "generateToken", returnType: CAPPluginReturnPromise),
    ]
    @objc func generateToken(_ call: CAPPluginCall) {       
        guard DCDevice.current.isSupported == true else {
            call.reject("DeviceCheck is not supported on this device")
            return
        }
        DCDevice.current.generateToken { tokenData, error in
            guard error == nil,
                  let deviceToken = tokenData?.base64EncodedString() else {
                switch error {
                case .none: call.reject("DeviceCheck token encoding failed")
                case .some(let error):
                    call.reject("DeviceCheck error:\(error.localizedDescription)")
                }
                return
            }
            call.resolve(["token": deviceToken])
        }
    }
}
